src/rgw/driver/rados/rgw_trim_bilog.cc

```diff
@@ -367,7 +367,7 @@ class BucketTrimShardCollectCR : public RGWShardCollectCR {
   size_t i{0}; //< index of current shard marker
 
   int handle_result(int r) override {
-    if (r == -ENOENT) { // ENOENT is not a fatal error
+    if (r == -ENODATA) { // ENODATA is not a fatal error
```

Comment:
i'm guessing this used ENOENT in case the shard object was deleted. but `RGWRadosBILogTrimCR` doesn't use `op.assert_exists()`, so a trim op would probably recreate the object in that case. would you be willing to add that, and update this check to ignore both ENOENT and ENODATA?

Reply:
just out of curiosity, what would be the case where the shard is not there, and we should still suppress the error and take it as ok?

Reply:
i can't think of any cases where we'd _expect_ to see this, but if someone runs `radosgw-admin bi purge` or `rados rm` to delete the shard objects, trim shouldn't recreate them

Reply:
Was more curious whether we should report the error for `ENOENT` - but I think other APIs will take it and should not be the trimming's job.

Reply:
good point. since it's unexpected, it probably is worth logging a warning there

Reply:
ok, i now have better wording for the commit :sweat_smile: 
Let me know if we need dedicated logging (perhaps prefixes with `WARN: `) for `ENOENT`.